cin_validator/rules/cin2022_23/rule_8920.py

```diff
@@ -0,0 +1,271 @@
+
+    df_merged["ERROR_ID"] = tuple(
+        zip(df_merged[LAchildID], df_merged[CPPendDate], df_merged[PersonDeathDate])
+    )
+
+    # The merges were done on copies of cpp_df and child_df so that the column names in dataframes themselves aren't affected by the suffixes.
+    # we can now map the suffixes columns to their corresponding source tables such that the failing ROW_IDs and ERROR_IDs exist per table.
+    df_cpp_issues = (
+        df_cpp.merge(df_merged, left_on="ROW_ID", right_on="ROW_ID_cpp")
+        .groupby("ERROR_ID")["ROW_ID"]
+        .apply(list)
+        .reset_index()
+    )
+    df_reviews_issues = (
+        df_child.merge(df_merged, left_on="ROW_ID", right_on="ROW_ID_reviews")
+        .groupby("ERROR_ID")["ROW_ID"]
+        .apply(list)
+        .reset_index()
+    )
+
+    # Ensure that you maintain the ROW_ID, and ERROR_ID column names which are shown above. They are keywords in this project.
+    rule_context.push_type_2(
+        table=ChildProtectionPlans, columns=[CPPendDate], row_df=df_cpp_issues
+    )
+    rule_context.push_type_2(
+        table=ChildIdentifiers, columns=[PersonDeathDate], row_df=df_reviews_issues
+    )
+
+
+def test_validate():
+    # Create some sample data such that some values pass the validation and some fail.
+    sample_cpp = pd.DataFrame(
+        [
+            {
+                "LAchildID": "child1",
+                "CINdetailsID": "CDID1",
+                "CPPendDate": "26/05/2000",  # Fails as PersonDeathDate is before the CPPendDate
+            },
+            {
+                "LAchildID": "child2",
+                "CINdetailsID": "CDID2",
+                "CPPendDate": "27/06/2002",  # Fails as PersonDeathDate is before the CPPendDate
+            },
+            {
+                "LAchildID": "child3",
+                "CINdetailsID": "CDID3",
+                "CPPendDate": "07/02/2001",  # Fails as PersonDeathDate is before the CPPendDate
+            },
+            {
+                "LAchildID": "child4",
+                "CINdetailsID": "CDID4",
+                "CPPendDate": "26/05/2000",  # Passes as PersonDeathDate is after the CPPendDate
+            },
+            {
+                "LAchildID": "child5",
+                "CINdetailsID": "CDID5",
+                "CPPendDate": "26/05/2000",  # Passes as PersonDeathDate is after the CPPendDate
+            },
+            {
+                "LAchildID": "child6",
+                "CINdetailsID": "CDID6",
+                "CPPendDate": pd.NA,  # Ignored as rows with no CPPendDate are dropped (this is picked up by other rules)
+            },
+            {
+                "LAchildID": "child7",
+                "CINdetailsID": "CDID7",
+                "CPPendDate": "14/03/2001",  # Ignored as rows with no PersonDeathDate are dropped (this is picked up by other rules)
+            },
+        ]
+    )
+    sample_children = pd.DataFrame(
+        [
+            {
+                "LAchildID": "child1",
```

Comment:
As child identifiers doesn't have a CINdetailsID column, this one shouldn't have one either!